abixen-platform-core/src/test/java/com/abixen/platform/core/service/UserServiceTest.java

```diff
@@ -58,6 +63,55 @@ public void createUser() {
         assertNotNull(userFromDB);
     }
 
+    @Test
+    public void changeUserPasswordPositiveCase() {
+        log.debug("changeUserPassword() positive case");
+        String newpassword = "newPassword";
+
+        UserBuilder userBuilder = domainBuilderService.newUserBuilderInstance();
+        userBuilder.credentials("usernameA", "password");
+        userBuilder.screenName("screenNameA");
+        userBuilder.personalData("firstName", "middleName", "lastName");
+        userBuilder.additionalData(new Date(), "jobTitle", UserGender.MALE);
+        userBuilder.registrationIp("127.0.0.1");
+        User user = userBuilder.build();
+        userService.createUser(user);
+
+        UserChangePasswordForm passwordForm = new UserChangePasswordForm();
+        passwordForm.setCurrentPassword("password");
+        passwordForm.setNewPassword(newpassword);
+
+        UserChangePasswordForm newPasswordForm = userService.changeUserPassword(user, passwordForm);
+        User userFromDB = userService.findUser("usernameA");
+
+        PasswordEncoder encoder = new BCryptPasswordEncoder();
+
+        assertNotNull(userFromDB);
+        assertTrue(encoder.matches(newpassword, userFromDB.getPassword()));
```

Comment:
as I am understood, main idea of method is store new password in DB. So my assumption that in positive case we need to read user object from DB after changing password execution, and compare it with new password from form.